[RPC] Support timeout in rref._get_type()

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50499 [RPC] Support timeout for RRef proxy functions
* **#50498 [RPC] Support timeout in rref._get_type()**

This change is mostly needed for the next diff in this stack, where
rref._get_type() is called in the rpc_async/rpc_sync RRef proxy function and
can block indefinitely if there is no timeout. It will also be useful to have a
timeout argument when we publicize this API to keep it consistent with other
RPC APIs.

Differential Revision: [D25897588](https://our.internmc.facebook.com/intern/diff/D25897588/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25897588/)!

rohan-varma committed:
[RPC] Support timeout in rref._get_type()

This change is mostly needed for the next diff in this stack, where
rref._get_type() is called in the rpc_async/rpc_sync RRef proxy function and
can block indefinitely if there is no timeout. It will also be useful to have a
timeout argument when we publicize this API to keep it consistent with other
RPC APIs.

Differential Revision: [D25897588](https://our.internmc.facebook.com/intern/diff/D25897588/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25897588/)!

[ghstack-poisoned]

pritamdamania87 commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -373,13 +373,22 @@ PyObject* rpc_init(PyObject* _unused, PyObject* noargs) {
+
+                  Args:
+                    rref (torch.distributed.rpc.RRef): The RRef to get type of.
+                    timeout (float, optional): Timeout for ``_get_type``. If
+                          the call does not complete within this timeframe, an
+                          exception indicating so will be raised. If this
+                          argument is not provided, the default RPC timeout will
+                           be used.
```
The doc should mention the unit for the timeout (seconds/milliseconds etc.)

rohan-varma committed:
Update on "[RPC] Support timeout in rref._get_type()"


This change is mostly needed for the next diff in this stack, where
rref._get_type() is called in the rpc_async/rpc_sync RRef proxy function and
can block indefinitely if there is no timeout. It will also be useful to have a
timeout argument when we publicize this API to keep it consistent with other
RPC APIs.

Differential Revision: [D25897588](https://our.internmc.facebook.com/intern/diff/D25897588/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25897588/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] Support timeout in rref._get_type()"


This change is mostly needed for the next diff in this stack, where
rref._get_type() is called in the rpc_async/rpc_sync RRef proxy function and
can block indefinitely if there is no timeout. It will also be useful to have a
timeout argument when we publicize this API to keep it consistent with other
RPC APIs.

Differential Revision: [D25897588](https://our.internmc.facebook.com/intern/diff/D25897588/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25897588/)!

[ghstack-poisoned]
